radicle-httpd/src/api/v1/projects.rs

```diff
@@ -230,14 +232,11 @@ async fn tree_handler(
 ) -> impl IntoResponse {
     let path = path.strip_prefix('/').ok_or(Error::NotFound)?.to_string();
     let storage = &ctx.profile.storage;
-    let repo = storage.repository(project)?;
-    let tree = radicle_surf::object::tree(&repo.raw().into(), Some(Sha { sha }), Some(path))?;
-    let response = json!({
-        "path": &tree.path,
-        "entries": &tree.entries,
-        "info": &tree.info,
-        "stats": stats(&repo)?,
-    });
+    let repo = Repository::open(paths::repository(storage, &project))?;
+    let path = if path == "" { None } else { Some(&path) };
+    let tree = Tree::new(&repo, &sha, path)?;
+    let mut response = json!(tree);
+    response["stats"] = json!(repo.stats()?);
```

Comment:
`Tree::new(..)` won't take an empty path so I have to do this. However the response from API has broken the interface.

JSON:
```
{
    "entries": [{
        "path": "",
        "name": "README",
        "lastCommit": null,
        "kind": "blob"
    }, {
        "path": "",
        "name": "test_dir",
        "lastCommit": null,
        "kind": "directory"
    }],
    "lastCommit": {
        "sha1": "4f30784e586d09c0fb8d6c38ea35d6fd255ae10a",
        "author": {
            "name": "xphoniex",
            "email": "xphoniex@users.noreply.github.com"
        },
        "summary": "Initial commit",
        "description": "",
        "committer": {
            "name": "xphoniex",
            "email": "xphoniex@users.noreply.github.com"
        },
        "committerTime": 1670943683
    },
    "name": "",
    "path": "",
    "stats": {
        "commits": 1,
        "branches": 1,
        "contributors": 1
    }
}
```

Browser:
![Screenshot from 2022-12-13 19-06-20](https://user-images.githubusercontent.com/14107758/207378472-6640a711-d47b-46da-9c66-a4f3f49c929e.png)

It's because both entries have the same path. What's the issue?

cc @FintanH

Reply:
Ah, was `"path"` the full file path originally? The change we made was that `path` was where the file was located and `name` was the name of the blob. So `full_path = path ++ name`. This can be changed in the serialize instance for `TreeEntry` on the surf side.

Reply:
I should be able to update the PR to the new hash once you change it on the surf side.